### DISCUSSION TITLE
Unable to open app config file

### DISCUSSION THREAD

**Discussion by everydaylearner** [ejsuncy/sense_energy_prometheus_exporter#50] [General]
Hi there,

I pulled the latest image and container started successfully but i am seeing the following in the logs. Am i missing something? 

2023-07-28 15:45:22 Log level set to: INFO
2023-07-28 15:45:22 App config file: /etc/sense_energy_prometheus_exporter/config.yaml
2023-07-28 15:45:22 Unable to open app config file /etc/sense_energy_prometheus_exporter/config.yaml: [Errno 2] No such file or directory: '/etc/sense_energy_prometheus_exporter/config.yaml'
2023-07-28 15:45:24 Starting server on 0.0.0.0:9993

is that expected or one time thing? 

When go to http://localhost:9993 i can see metrics
Sample:
 TYPE sense_energy_trend_kWh gauge
sense_energy_trend_kWh{account="52home",period="daily",type="consumption"} 23.250935
sense_energy_trend_kWh{account="52home",period="daily",type="production"} 0.0
sense_energy_trend_kWh{account="52home",period="weekly",type="consumption"} 149.44484
sense_energy_trend_kWh{account="52home",period="weekly",type="production"} 0.0
sense_energy_trend_kWh{account="52home",period="monthly",type="consumption"} 1055.0557
sense_energy_trend_kWh{account="52home",period="monthly",type="production"} 0.0
sense_energy_trend_kWh{account="52home",period="yearly",type="consumption"} 6429.5684
sense_energy_trend_kWh{account="52home",period="yearly",type="production"} 0.0

Thank you!

**Comment by ejsuncy** [comment]
This error message just means that you didn't mount a config file, which is fine if you don't need to override the default configs in the app. I've pushed some changes to classify the logs as warnings instead of errors so it's not as confusing. If you pull tag `0.7.0a` you should pick up the changes, although you should be fine running your current image hash.

**Comment by everydaylearner** [comment]
thank you for all the updates! Do you mind putting a sample config file?

I see you have the following example but its more geared to kubernetes 

apiVersion: v1
data:
  prometheus_config: |
    global:
      scrape_interval: 15s
      external_labels:
        monitor: 'k8s-prom-monitor'
    scrape_configs:
      - job_name: 'sense'
        metrics_path: /metrics
        scrape_interval: 3m
        scrape_timeout: 1m
        static_configs:
          - targets:
              - 'prometheus-exporters-sense'
kind: ConfigMap
metadata:
  name: prometheus-config

once the config file is created do you put under volumes:/etc//etc/sense_energy_prometheus_exporter/  folder for docker-compose? 

also what are default values if there is no config file found? 

Thank you!

**Reply by ejsuncy** [reply]
There is already a sample config file in the repository at [volumes/config.yaml](https://github.com/ejsuncy/sense_energy_prometheus_exporter/blob/main/volumes/config.yaml). Currently the only config option is the exporter namespace. The default is `sense_energy`, which is the prefix for the metric names (see [data/metrics.txt](https://github.com/ejsuncy/sense_energy_prometheus_exporter/blob/main/data/metrics.txt) for the metric names that are produced). If you don't need to override this namespace, you can omit the config file entirely without issues.

**Reply by ejsuncy** [reply]
If you do want to override the namespace via the config file, see the existing [docker-compose.yaml file](https://github.com/ejsuncy/sense_energy_prometheus_exporter/blob/fa34a3dbc9fe3fc535da3be5c4c16318b3036ee1/docker-compose.yaml#L15) for how the config file volume should be mounted.

For running directly with docker, the readme docker command has the correct parameter for mounting a volume as well.
